### ISSUE TITLE
Authoritative creates NSEC3 recode for glue recode

### ISSUE THREAD

**Issue by mimuret** [PowerDNS/pdns#8892]
- Program: Authoritative
 - Issue type: Bug report

### Short description
Authoritative creates NSEC3 recode for glue recode.

https://tools.ietf.org/html/rfc4034#section-4.1.1

```
   Owner names of RRsets for which the given zone is not authoritative
   (such as glue records) MUST NOT be listed in the Next Domain Name
   unless at least one authoritative RRset exists at the same owner
   name.
```

### Environment

 - Operating system: Centos7
 - Software version: 4.2
 - Software source: compiled yoursel

### Steps to reproduce

1. 
2. 
3.

**Comment by Habbie** [comment]
Can you please show us what is happening?

**Comment by mimuret** [comment]
I use pdns for hidden master server.
I set pre-signed zone.

I signed zone using dnssec-signzone. 
it not create glue name NSEC3 and RRSIG.

I look like that pdns creates NSEC3 glue recode.

This zone fail verify denial of existence.

AXFR LOG
https://gist.github.com/mimuret/83105fa5b49349b9580063b5715a68a0

I think presign zone must run rectify function (tcpreceiver.cc).

**Comment by Habbie** [comment]
Indeed, your log shows an NSEC3 chain including `ns.sub` and `sub`, and shows that there is no RRSIG for the `ns.sub` one.

How are you loading the zone into PowerDNS? What backend?

**Comment by mind04** [comment]
Setting presigned AND running dnssec-signzone may lead to unexpected results. You cannot use them together.

**Comment by mimuret** [comment]
> Indeed, your log shows an NSEC3 chain including `ns.sub` and `sub`, and shows that there is no RRSIG for the `ns.sub` one.
> 
Yes, because glue record MUST NOT add NSEC/NSEC3 recorde.

> How are you loading the zone into PowerDNS? What backend?
I use gmysql backend loading by custom app using SQL.

It is reproducible using zone2sql.
https://github.com/mimuret/pdns-nsec3

**Comment by Habbie** [comment]
Thank you for the reproduction, it will be helpful.

I spent some time discussing this, and the honest answer is: `zone2sql` and `pdnsutil load-zone` are not currently suitable for presigned zones. The docs about this are wrong.

If you want to get this working now, we suggest slaving the zone from BIND or NSD on localhost. We will think about how we can improve this situation on the PowerDNS side.

**Comment by Habbie** [comment]
I've updated the ticket title to reflect this.
